src/main/java/pl/edu/agh/firecell/engine/algorithm/FirePropagator.java

```diff
@@ -90,34 +86,31 @@ private int computeBurningTimeWood(State oldState, Vector3i cellIndex, double ne
         Cell oldCell = oldState.getCell(cellIndex);
         int newBurningTime = currenBurningTime;
         if (oldCell.burningTime() == 0 &&
-                newTemperature > WOOD.autoIgnitionTemperature() &&
                 MAX_BURNING_TIME != 0 &&
                 oldCell.flammable()) {
-            newBurningTime++;
-        }
-        if (oldCell.burningTime() == 0 &&
-                newTemperature > WOOD.ignitionTemperature() &&
-                hasNeighbourLongOnFire(oldState, cellIndex) &&
-                MAX_BURNING_TIME != 0 &&
-                oldCell.flammable()) {
-            newBurningTime++;
+            if (newTemperature > Material.WOOD.autoIgnitionTemperature()) {
+                newBurningTime++;
+            } else if (shouldIgniteFromNeighbour(oldState, cellIndex) &&
+                    newTemperature > Material.WOOD.ignitionTemperature()) {
+                newBurningTime++;
+            }
         }
         if (oldCell.burningTime() > 0 && oldCell.burningTime() <= MAX_BURNING_TIME) {
             newBurningTime++;
         }
         return newBurningTime;
     }
 
-    private boolean hasNeighbourLongOnFire(State oldState, Vector3i cellIndex){
+    private boolean shouldIgniteFromNeighbour(State oldState, Vector3i cellIndex) {
         int requiredTime = 10;
-        return NeighbourUtils.neighboursStream(cellIndex, X)
+        return NeighbourUtils.neighboursStream(cellIndex, NeighbourUtils.Axis.X)
                 .filter(oldState::hasCell)
```

Comment:
Let's concatenate those streams instead of duplicating code, since they are connected with `or` and both uses `anyMatch()`
```suggestion
               return Stream.concat(
                        NeighbourUtils.neighboursStream(cellIndex, NeighbourUtils.Axis.X),
                        NeighbourUtils.neighboursStream(cellIndex, NeighbourUtils.Axis.Z)
                ).filter(oldState::hasCell)
                .map(oldState::getCell)
                .anyMatch(cell -> cell.burningTime() > requiredTime);
```